tests/test_gen_utils.py

```diff
@@ -0,0 +1,86 @@
+import os
+
+import pytest
+import ndmg
+from ndmg.utils.cloud_utils import s3_get_data
+from pathlib import Path
+from ndmg.utils.gen_utils import create_datadescript, DirectorySweeper
+
+
+@pytest.fixture
+def input_dir_tree(tmp_path):
+    data = {"002547": [1,2], "002548": [1], "002449": [1,2,3]}
+    for sub, session in data.items():
+        for ses in session:
+            info = f"sub-{sub}/ses-{ses}"
+            anat = os.path.join(tmp_path, info, "anat")
+            dwi = os.path.join(tmp_path, info, "dwi")
+            # make directories and files
+            os.makedirs(anat)
+            os.makedirs(dwi)
+            tmpfilepath = os.path.join(anat, f"sub-{sub}_ses-{ses}_T1w.nii.gz")
+            with open(tmpfilepath, "x") as f:
+                f.write("placeholder text")
+            tmpfilepath = os.path.join(dwi, f"sub-{sub}_ses-{ses}_dwi.bval")
+            with open(tmpfilepath, "x") as f:
+                f.write("placeholder text")
+            tmpfilepath = os.path.join(dwi, f"sub-{sub}_ses-{ses}_dwi.bvec")
+            with open(tmpfilepath, "x") as f:
+                f.write("placeholder text")
+            tmpfilepath = os.path.join(dwi, f"sub-{sub}_ses-{ses}_dwi.nii.gz")
+            with open(tmpfilepath, "x") as f:
+                f.write("placeholder text")
```

Comment:
We can likely do something to avoid copy/pasting code here, but maybe not worth the time

Reply:
I toyed around with a way to simplify it, but it was a pain and would add yet another layer to the for loop. I know it's not the most efficient, but I kind of prefer it this way for readability.